compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -478,6 +479,44 @@ fn check_opaque_meets_bounds<'tcx>(
     let _ = infcx.inner.borrow_mut().opaque_type_storage.take_opaque_types();
 }
 
+fn is_enum_of_nonnullable_ptr<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    adt_def: AdtDef<'tcx>,
+    substs: SubstsRef<'tcx>,
+) -> bool {
+    if adt_def.repr().inhibit_enum_layout_opt() {
+        return false;
+    }
+
+    if let Some(ty) = match &adt_def.variants().raw[..] {
+        [var_one, var_two] => match (&var_one.fields[..], &var_two.fields[..]) {
+            ([], [field]) | ([field], []) => Some(field.ty(tcx, substs)),
+            _ => None,
+        },
+        _ => None,
+    } {
```

Comment:
Use let else and an early exit to reduce the indentation a bit, e.g.:

```rust
let [var_one, var_two] = &adt_def.variants().raw[..] else {
    return false;
};
let (([], [field]) | ([field], [])) = (&var_one.fields[..], &var_two.fields[..]) else {
    return false;
};
```

Reply:
Done